daffodil-propgen/src/main/scala/org/apache/daffodil/propGen/PropertyGenerator.scala

```diff
@@ -430,41 +430,44 @@ trait CurrencyMixin extends PropertyMixin {
 
 """
 
+  private val expressionAllowedProperties = List(
+    "byteOrder",
+    "encoding",
+    "initiator",
+    "terminator",
+    "outputNewLine",
+    "length",
+    "escapeCharacter",
+    "escapeEscapeCharacter",
+    "textStandardDecimalSeparator",
+    "textStandardGroupingSeparator",
+    "textStandardExponentRep",
+    "binaryFloatRep",
+    "textBooleanTrueRep",
+    "textBooleanFalseRep",
+    "separator",
+    "occursCount",
+    "inputValueCalc",
+    "outputValueCalc",
+    "calendarLanguage",
+    "choiceDispatchKey"
+  )
+
   /**
    * exclude these since code should use the CompiledExpression created from these values,
    * not the property values themselves. We'll do these by hand.
    */
   def excludeRuntimeProperties(propName: String) = {
     val runtimeValuedProperties = List(
-      "byteOrder",
-      "encoding",
-      "initiator",
-      "terminator",
-      "outputNewLine",
-      "length",
-      "escapeCharacter",
-      "escapeEscapeCharacter",
-      "textStandardDecimalSeparator",
-      "textStandardGroupingSeparator",
-      "textStandardExponentRep",
-      "binaryFloatRep",
-      "textBooleanTrueRep",
-      "textBooleanFalseRep",
-      "separator",
-      "occursCount",
-      "inputValueCalc",
-      "outputValueCalc",
       "textStandardInfinityRep",
       "textStandardNaNRep",
       "textStandardZeroRep",
       "nilValue",
       "textStringPadCharacter",
       "textNumberPadCharacter",
       "textBooleanPadCharacter",
```

Comment:
The ones now in the `runtimeValuedProperties` are not really runtime valued properties, they are just done by hand for some reason (i'm not sure why?). I wonder if they really belong in the `excludedBecauseDoneByHand` member up above? And then `excludeRuntimeProperties` really only has to look at `expressionAllowedProperities`